Assets/_Scripts/Actions/PoisonAllTest.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "PoisonAllTest", menuName = "Actions/Poison All", order = 2)]
+public class PoisonAllTest : ActionBase {
+    [SerializeField]
+    private List<StatusAilment> _status;
+
+    public override bool TryChooseAction(FightingEntity user, string[] splitCommand) {
+        // Poison All command format: !m poison2
+        if (!BasicValidation(splitCommand)) {
+            return false;
+        }
+        List<int> targetIds = this.GetAllPossibleTargets(user).Map((FightingEntity target) => target.targetId );
+        user.SetQueuedAction(new QueuedAction(user, this, targetIds));
+        return true;
+    }
+
+    public override FightResult ApplyEffect(FightingEntity user, List<FightingEntity> targets) {
+        PlayerStats before, after;
```

Comment:
Doesn't have to be done in this pr, but can you think of a way to add helper functions so we don't have to rewrite this code for every action?